onfido-java/src/test/java/com/onfido/integration/CheckManagerTest.java

```diff
@@ -1,222 +1,141 @@
+    assertRequestField("sub_result", "rejected");
 
-    Check check =
-        onfido.check.create(Check.request().applicantId("id").reportNames("document", "identity_enhanced").consider("identity_enhanced"));
+    assertEquals(applicant.getId(), check.getApplicantId());
+  }
 
-    // Correct path
-    RecordedRequest request = server.takeRequest();
-    assertEquals("/checks/", request.getPath());
+  @Test
+  public void createConsiderCheckTest() throws Exception {
+    Check check = createCheck(applicant, document,
+      Check.request().reportNames("document", "identity_enhanced")
+                     .consider("identity_enhanced"));
 
-    // Correct request body
-    String json = request.getBody().readUtf8();
-    JsonObject jsonObject = JsonObject.parse(json);
-    assertEquals("id", jsonObject.get("applicant_id"));
-    ArrayList<String> expectedConsiderArrayList = new ArrayList<>();
-    expectedConsiderArrayList.add("identity_enhanced");
-    assertEquals(expectedConsiderArrayList, jsonObject.get("consider"));
+    assertRequestField("applicant_id", applicant.getId());
+    assertRequestField("consider", List.of("identity_enhanced"));
 
-    // Correct response body
-    assertEquals("id", check.getApplicantId());
+    assertEquals(applicant.getId(), check.getApplicantId());
   }
 
   @Test
-  public void createDrivingLicenceCheck() throws Exception {
-    String response =
-            new JsonObject()
-                    .add("applicant_id", "id")
-                    .toJson();
-
-    MockWebServer server = mockRequestResponse(response);
-
-    Onfido onfido =
-            Onfido.builder().apiToken("token").unknownApiUrl(server.url("/").toString()).build();
-
-    Check check =
-            onfido.check.create(Check.request()
-                    .applicantId("id")
-                    .reportNames("us_driving_licence")
-                    .usDrivingLicence(DrivingLicence.request().idNumber("12345").stateCode("GA"))
-            );
-
-    // Correct path
-    RecordedRequest request = server.takeRequest();
-    assertEquals("/checks/", request.getPath());
-
-    // Correct request body
-    String json = request.getBody().readUtf8();
-    JsonObject jsonObject = JsonObject.parse(json);
-    assertEquals("id", jsonObject.get("applicant_id"));
-
-    // Correct response body
-    assertEquals("id", check.getApplicantId());
+  public void createDrivingLicenceCheckTest() throws Exception {
+    if (isMockingEnabled()) {
+      Check check = createCheck(applicant, document,
+        Check.request().reportNames("document")
+                      .consider("identity_enhanced")
+                      .reportNames("us_driving_licence")
+                      .usDrivingLicence(DrivingLicence.request().idNumber("12345")
+                                                                .state("GA")));
+
+      assertRequestField("applicant_id", applicant.getId());
+
+      assertEquals(applicant.getId(), check.getApplicantId());
+    }
   }
```

Comment:
I suggest we use `throw new SkipException(String message)` to skip a test 💅 


For example 👇 

```suggestion
  public void createDrivingLicenceCheckTest() throws Exception {
    if (!isMockingEnabled()) {
     throw new SkipException("Skipping createDrivingLicenceCheckTest() because mocking is not enabled.");
    }
    
    ...
  }
```

Reply:
To keep code lighter, I propose to use this brand new `skipTestIfMockingNotEnabled()` helper function (anyway it looks like Exception message is not logged anywhere). What do you think about?